addon/mixins/base-query-manager.js

```diff
@@ -0,0 +1,24 @@
+import Ember from 'ember';
+import QueryManager from 'ember-apollo-client/apollo/query-manager';
+
+const { getOwner, Mixin } = Ember;
+
+export default Mixin.create({
+  apollo: null,
+
+  init() {
+    this._super(...arguments);
+    let owner = getOwner(this);
+
+    // I think this is necessary in order to make this register properly when used in
+    // unit or integration tests, but I'm not sure it's the right way to solve it.
+    if (!owner.hasRegistration('object:apollo-query-manager')) {
+      owner.register('object:apollo-query-manager', QueryManager, {
+        singleton: false,
+      });
+    }
```

Comment:
I have no idea what I'm doing here, but I had to figure something out to make this work reasonably well in unit + integration tests for routes/components where the mixins were being used. Without this section, `apollo` was undefined in routes and components.